base/interactiveutil.jl

```diff
@@ -64,7 +64,27 @@ function edit(path::AbstractString, line::Integer=0)
     nothing
 end
 
-edit(f)          = edit(functionloc(f)...)
+function edit(f)
+    ms = methods(f)
```

Comment:
It also (non-quotes) works on, say, types. The new definition just replaces one error branch with an interactive choice.